wrappers/services_test.go

```diff
@@ -445,6 +445,55 @@ ListenStream=%s
 	c.Check(sock2File, testutil.FileContains, expected)
 }
 
+func (s *servicesTestSuite) TestAddSnapUserSocketFiles(c *C) {
+	info := snaptest.MockSnap(c, packageHello+`
+ svc1:
+  daemon: simple
+  daemon-mode: user
+  plugs: [network-bind]
+  sockets:
+    sock1:
+      listen-stream: $SNAP_USER_COMMON/sock1.socket
+      socket-mode: 0666
```

Comment:
I wonder if we should limit what the socket-mode should be with user services. On Ubuntu (at least), home directories are 0755 and ~/snap is world readable, meaning that that SNAP_USER_DATA and SNAP_USER_COMMON are too, so a snap could set this inappropriately to 0666 which would allow another user on the system to mess with this user's snap user services. This is perhaps best left for the review-tools.

Reply:
That's a fair point for snapd to not do anything with it. I'm still considering an update for the review-tools.